src/lib.rs

```diff
@@ -341,6 +341,8 @@ pub fn optimize(input: &InFile, output: &OutFile, opts: &Options) -> PngResult<(
         eprintln!("Processing: {}", input);
     }
 
+    let deadline = Arc::new(Deadline::new(opts.timeout, opts.verbosity.is_some()));
```

Comment:
Does it make more sense to instantiate the Deadline (here and on 443) inside `optimize_png` to avoid the duplication?

Reply:
It starts counting time from when it's instantiated, and I wanted it to also include time it takes to load the image.